src/pageEditor/hooks/useCreateModFromUnsavedMod.ts

```
@@ -0,0 +1,199 @@
+  const client = await getLinkedApiClient();
+  const data = { config: objToYaml(config), kind: "extensionPoint" };
+  await client.put(API_PATHS.BRICK(packageUuid), data);
+}
+
+type UseCreateModFromUnsavedModReturn = {
+  createModFromUnsavedMod: (modMetadata: ModMetadataFormState) => Promise<void>;
+};
+
+function useCreateModFromUnsavedMod(
+  unsavedModId: Nullishable<RegistryId>,
+): UseCreateModFromUnsavedModReturn {
+  const dispatch = useDispatch();
+  const [createMod] = useCreateModDefinitionMutation();
+  const { data: editablePackages } = useGetEditablePackagesQuery();
+  const { buildAndValidateMod } = useBuildAndValidateMod();
+  const getCleanComponentsAndDirtyFormStatesForMod = useSelector(
+    selectGetCleanComponentsAndDirtyFormStatesForMod,
+  );
+  const { cleanModComponents, dirtyModComponentFormStates } = useMemo(
+    () => getCleanComponentsAndDirtyFormStatesForMod(unsavedModId ?? null),
+    [getCleanComponentsAndDirtyFormStatesForMod, unsavedModId],
+  );
+  const dirtyModOptionsById = useSelector(selectDirtyModOptionsDefinitions);
+  const dirtyModOptionsDefinition = useMemo(
+    // eslint-disable-next-line security/detect-object-injection -- Registry id
+    () => (unsavedModId ? dirtyModOptionsById[unsavedModId] : undefined),
+    [dirtyModOptionsById, unsavedModId],
+  );
+
+  const createModFromUnsavedMod = useCallback(
+    (
+      modMetadata: ModMetadataFormState,
+      // eslint-disable-next-line @typescript-eslint/promise-function-async -- permissions check must be called in the user gesture context, `async-await` can break the call chain
+    ) =>
+      ensureModComponentFormStatePermissionsFromUserGesture(
+        dirtyModComponentFormStates,
+        // eslint-disable-next-line promise/prefer-await-to-then -- permissions check must be called in the user gesture context, `async-await` can break the call chain
+      ).then(async (hasPermissions) => {
+        if (!hasPermissions || !unsavedModId || !editablePackages) {
```

Would we want to throw an error if there is no unsavedModId? From what I understand something wrong would be going on if that were the case

---

yep good call we can actually make this required now